metricsaggregator/src/main/java/io/dockstore/metricsaggregator/helper/ExecutionTimeAggregator.java

```diff
@@ -0,0 +1,116 @@
+ */
+public final class ExecutionTimeAggregator implements RunExecutionAggregator<ExecutionTimeMetric, String> {
+    @Override
+    public String getMetricFromRunExecution(RunExecution runExecution) {
+        return runExecution.getExecutionTime();
+    }
+
+    @Override
+    public ExecutionTimeMetric getMetricFromMetrics(Metrics metrics) {
+        return metrics.getExecutionTime();
+    }
+
+    @Override
+    public Optional<RunExecution> getWorkflowExecutionFromTaskExecutions(TaskExecutions taskExecutionsForOneWorkflowRun) {
+        final List<RunExecution> taskExecutions = taskExecutionsForOneWorkflowRun.getTaskExecutions();
+        if (taskExecutions != null && taskExecutions.stream().map(RunExecution::getExecutionTime).allMatch(Objects::nonNull)) {
+            // We cannot calculate the overall total time from RunExecution's executionTime, which is in ISO 8601 duration format.
+            // Calculate a best guess using RunExecution's dateExecuted, which is in ISO 8601 date format
+            if (taskExecutions.size() == 1 && taskExecutions.get(0).getExecutionTime() != null) {
+                // If there's only one task, set the workflow-level execution time to be the execution time of the single task
+                return Optional.of(new RunExecution().executionTime(taskExecutions.get(0).getExecutionTime()));
+
+            // Calculate a duration if all task executions have a valid dateExecuted
+            } else if (taskExecutions.stream().allMatch(execution -> checkExecutionDateISO8601Format(execution.getDateExecuted()).isPresent())) {
+                // Find the earliest date executed and latest date executed to calculate a duration estimate
+                final Optional<Date> earliestTaskExecutionDate = taskExecutions.stream()
+                        .map(execution -> checkExecutionDateISO8601Format(execution.getDateExecuted()).get())
+                        .min(Date::compareTo);
+                final Optional<Date> latestTaskExecutionDate = taskExecutions.stream()
+                        .map(execution -> checkExecutionDateISO8601Format(execution.getDateExecuted()).get())
+                        .max(Date::compareTo);
+                final Optional<RunExecution> latestTaskExecuted = taskExecutions.stream()
+                        .max(Comparator.comparing(execution -> checkExecutionDateISO8601Format(execution.getDateExecuted()).get(), Date::compareTo));
+
+                if (earliestTaskExecutionDate.isPresent() && latestTaskExecutionDate.isPresent() && latestTaskExecuted.isPresent()) {
+                    // Execution dates are the start dates, calculate a rough duration from the execution dates of the earliest and latest tasks
+                    long durationInMs = latestTaskExecutionDate.get().getTime() - earliestTaskExecutionDate.get().getTime();
+                    Duration duration = Duration.of(durationInMs, ChronoUnit.MILLIS);
```

Comment:
FWIW, I can't imagine people would care much about anything more granular than minutes. Assuming the most granular billing is per minute. 

That said, this probably  gets converted for display anyway